modules/grizzly/src/main/java/org/glassfish/grizzly/nio/RoundRobinConnectionDistributor.java

```diff
@@ -155,17 +167,17 @@ public SelectorRunner nextService() {
             return next();
         }
     }
-    
+
     private final class ServDedicatedIterator implements Iterator {
         private int counter;
```

Comment:
why are you not using an AtomicInteger in this case compared to the other iterator?